src/proxysql.rs

```diff
@@ -0,0 +1,249 @@
+            messages::print_info("Inserted warm-up rule");
+        } else {
+            self.conn.query_drop(format!("INSERT INTO mysql_query_rules (username, destination_hostgroup, active, digest, apply, comment) VALUES ('{}', {}, 1, '{}', 1, '{}: {}')", query.get_user(), self.readyset_hostgroup, query.get_digest(), DESTINATION_QUERY_TOKEN, date_formatted)).expect("Failed to insert into mysql_query_rules");
+            messages::print_info("Inserted destination rule");
+        }
+        Ok(true)
+    }
+
+    pub fn load_query_rules(&mut self) -> Result<bool, mysql::Error> {
+        self.conn
+            .query_drop("LOAD MYSQL QUERY RULES TO RUNTIME")
+            .expect("Failed to load query rules");
+        Ok(true)
+    }
+    pub fn save_query_rules(&mut self) -> Result<bool, mysql::Error> {
+        self.conn
+            .query_drop("SAVE MYSQL QUERY RULES TO DISK")
+            .expect("Failed to load query rules");
+        Ok(true)
+    }
+
+    /// This function is used to check the current list of queries routed to Readyset.
+    ///
+    /// # Arguments
+    /// * `conn` - A reference to a connection to ProxySQL.
+    ///
+    /// # Returns
+    /// A vector of tuples containing the digest_text, digest, and schemaname of the queries that are currently routed to ReadySet.
+    pub fn find_queries_routed_to_readyset(&mut self) -> Vec<String> {
+        let rows: Vec<String> = self
+            .conn
+            .query(format!(
+            "SELECT digest FROM mysql_query_rules WHERE comment LIKE '{}%' OR comment LIKE '{}%'",
+            MIRROR_QUERY_TOKEN, DESTINATION_QUERY_TOKEN
+        ))
+            .expect("Failed to find queries routed to ReadySet");
+        rows
+    }
+
+    /// This function is used to check if any mirror query rule needs to be changed to destination.
+    ///
+    /// # Returns
+    ///
+    /// A boolean indicating if any mirror query rule was changed to destination.
+    pub fn adjust_mirror_rules(&mut self) -> Result<bool, mysql::Error> {
+        let mut updated_rules = false;
+        let datetime_now: DateTime<Local> = Local::now();
+        let tz = datetime_now.format("%z").to_string();
+        let date_formatted = datetime_now.format("%Y-%m-%d %H:%M:%S");
+        let rows: Vec<(u16, String)> = self.conn.query(format!("SELECT rule_id, comment FROM mysql_query_rules WHERE comment LIKE '{}: ____-__-__ __:__:__';", MIRROR_QUERY_TOKEN)).expect("Failed to select mirror rules");
+        for (rule_id, comment) in rows {
+            let datetime_mirror_str = comment
+                .split("Mirror by readyset scheduler at:")
+                .nth(1)
+                .unwrap_or("")
+                .trim();
+            let datetime_mirror_str = format!("{} {}", datetime_mirror_str, tz);
+            let datetime_mirror_rule =
+                DateTime::parse_from_str(datetime_mirror_str.as_str(), "%Y-%m-%d %H:%M:%S %z")
```

Comment:
This is the time zone decoding line I'm talking about.

Reply:
ah, ok - `parse_from_str` requires a TZ and we use localTZ.

Reply:
Ah I see.  I missed the part where we append the time zone to the string after we fetch it from ProxySQL.